src/checkers/inference/model/ComparisonConstraint.java

```diff
@@ -0,0 +1,140 @@
+package checkers.inference.model;
+
+import java.util.Arrays;
+
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.BugInCF;
+
+import com.sun.source.tree.Tree.Kind;
+
+/**
+ * Represents a constraint that two slots must be comparable.
```

Comment:
How about changing doc to:

Represents a constraint that two slots with numeric underlying types are compared through comparison operators.